Add MaxBytesReader for ocsp-responder.

jsha opened the pull request:
Also, return status code 500 when the OCSP response from
the DB is unparseable.

rolandshoemaker commented on ocsp/responder.go:
```diff
@@ -146,14 +146,33 @@ var responseTypeToString = map[ocsp.ResponseStatus]string{
+		prometheus.HistogramOpts{
+			Name:    "ocsp_request_sizes",
+			Help:    "Size of OCSP requests",
+			Buckets: []float64{1, 100, 200, 400, 800, 1200, 2000, 5000, 1000},
```
s/1000/10000/

rolandshoemaker commented on cmd/ocsp-responder/main.go:
```diff
@@ -326,8 +317,8 @@ func (om *ocspMux) Handler(_ *http.Request) (http.Handler, string) {
 
-func mux(stats prometheus.Registerer, responderPath string, source bocsp.Source, responseTypesCounter *prometheus.CounterVec) http.Handler {
-	stripPrefix := http.StripPrefix(responderPath, bocsp.NewResponder(source, responseTypesCounter))
+func mux(stats prometheus.Registerer, responderPath string, source bocsp.Source, stats prometheus.Registerer) http.Handler {
```
Using `stats` twice here.

jsha commented:
Added the stats. In the processes I had to move the unittests for the existing responseType stats over to the ocsp package, which led me to discover that (a) the existing unittest was hitting a ServerInternal error case because the mock `testSource` was returning bytes that are not valid ASN.1, and (b) the ServerInternal error case was incorrectly returning 200 due to a missing call to WriteHeader.

Fixed the mock, and fixed the status code for this particular error.
